### PR TITLE
[IMP] website: add connector options to step snippet

### PR DESCRIPTION
**Pull request by younn-o**
New options are added to the steps snippets, allowing to pick the type
of the steps connector and its color.

A Widget linked to the s_process_steps snippet is introduced to reload
the connectors when a column or the window is resized.
As the length and position of the connectors is computed from the
widget, a class o_loaded was added to avoid flickering before the final
render.
The s_process_steps snippet CSS version is now 001.

Coming from https://github.com/odoo/odoo/pull/71152

task-2463009

Co-authored-by: Matthieu Stockbauer <tsm@odoo.com>

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
@younn-o I push-forced this one and rebased it on master. Just wanted to fix the "tab" characters that were polluting the files before testing and hopefully merging for the release as a last minute small improvement... but the branch was just not tested at all, it does not work :disappointed: 

As soon as you resize the columns, this is broken :disappointed: 

In any case, if we want it today or tomorrow for the 15.0 release if you manage to adapt it until then, let's still target master: there is a new view in there so the runbot won't let us merge this unless it is first merged in master then backported.

Reminder: don't forget to fetch first as I push-forced :+1:

**Review by bso-odoo — COMMENTED** [review]
Here are two tiny comments.
Also, not directly related to this PR, I get a never-ending loader if I try to reduce the number of step columns.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
I fear this is a complex task, check my comments. Maybe it is worth reconsidering the spec as it seems to generate a lot of code for a not-so-huge improvement (don't hesitate to question the spec during dev if you see this is getting too complicated). Maybe the easiest solution would be to prevent column resizing when using something different than the straight line, I don't know. But +300LOC with code visitors have to load and run just to have an arrow between two columns... that seems too much :grimacing: 

Check with @pierre-pde once he has the time :wink:

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

**Review by qsm-odoo — APPROVED** [review]
I rebased, tested (looks really nice :+1:), fixed the commit message listed apps and also simplified a bit a switch case and details: https://github.com/odoo/odoo/compare/0f0276fe1c3ea62303bc9ba4ac05122ecdfac02c..d383970540d06f455446c6a67886edfd52834c21

@robodoo r+

**Pull request closed without merging**
